src/components/delivery/Compass.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import { connect } from "react-redux";
+
+const Compass = (props) => {
+  const { orientW, orientX, orientY, orientZ } = props;
+  // Calculate the direction based on the quaternion values
+  const direction = Math.round(
+    (Math.atan2(
+      2 * (orientW * orientZ + orientX * orientY),
+      1 - 2 * (orientY * orientY + orientZ * orientZ)
+    ) *
+      180) /
+      Math.PI
+  );
+
+  const roll = Math.round(
+    (Math.atan2(
+      2 * (orientW * orientX + orientY * orientZ),
+      1 - 2 * (orientX * orientX + orientY * orientY)
+    ) *
+      180) /
+      Math.PI
+  );
+  
+  const pitch = Math.round(
+    2 * (Math.atan2(
+      Math.sqrt(1 + 2 * (orientW * orientY - orientX * orientZ)),
+      Math.sqrt(1 - 2 * (orientW * orientY - orientX * orientZ))
+    )
+    ) - Math.PI / 2
+  );
+  let needleColor = "black";
+  if ( (roll >= -45 && roll <= 45) || (pitch >= -45 && pitch <= 45) ) {
+    needleColor = "green";
+  } else if ( (roll > 45 && roll <= 135) || (pitch > 45 && pitch <= 135) ) {
+    needleColor = "yellow";
+  } else {
+    needleColor = "red";
+  }
```

Comment:
Instead of thresholding separately on pitch and roll, can we turn it into one number (the magnitude of the rotation) and threshold based on that? Look up some math, there should be resources online on how to do it. If you're stuck, lmk and I can help out.